contract/r/gnoswap/v1/pool/assert.gno

```diff
@@ -52,3 +54,12 @@ func assertAmountSpecifiedIsNotZero(amountSpecified string) {
 		))
 	}
 }
+
+func assertPayerIsPreviousRealmOrOriginCaller(payer std.Address) {
+	if payer != std.PreviousRealm().Address() && payer != std.OriginCaller() {
+		panic(newErrorWithDetail(
+			errInvalidPayer,
+			ufmt.Sprintf("expected payer(%s) to be the previous realm or the caller", payer),
```

Comment:
to increase debuggability, it would be good to also log the prev realm or caller's address in the error message